ubuntu-image

```diff
@@ -1,2 +1,28 @@
 #!/bin/sh
+
+if [ -n "$SNAP" ]; then
+    # If we're running as a snap, do some additional wrapping for classic
+    case $SNAP_ARCH in
+    amd64)
+        TRIPLET="x86_64-linux-gnu"
+        ;;
+    armhf)
+        TRIPLET="arm-linux-gnueabihf"
+        ;;
+    arm64)
+        TRIPLET="aarch64-linux-gnu"
+        ;;
+    *)
+        TRIPLET="$(uname -p)-linux-gnu"
+        ;;
+    esac
+
+    # We need to set the library paths for our snap to find the included libs.
+    export LD_LIBRARY_PATH=$SNAP/usr/lib:$SNAP/usr/lib/$TRIPLET:$SNAP/lib/$TRIPLET
+    export PYTHONPATH=$SNAP/lib/python3.5/site-packages:$PYTHONPATH
+    # Since we use some system-owned tools like mount, sudo or snap that we can't
```

Comment:
"if you build the snap on xenial" - well, you're meant not to do that and we should make sure we aren't doing that anywhere.

I don't know about forcing minimum versions is snapcraft.yaml.  Seems like a good feature to have, if it doesn't exist.